### DISCUSSION TITLE
does qdrant uses post filtering or pre filtering

### DISCUSSION THREAD

**Discussion by vihariazure** [qdrant/qdrant#5706] [Q&A] [answered]
hi team i came across below things in the qdrant

Qdrant does not primarily use either **pre-filtering or post-filtering**; instead, it employs a unique approach that allows for filtering directly within the vector search process, avoiding the need for separate filtering stages, making it faster and more accurate compared to traditional pre or post filtering methods.

the question is does qdrant uses filtering or not. if it uses which filtering post or pre

thanks in advance

**Comment by generall** [comment, marked as answer]
Neither: https://qdrant.tech/articles/filtrable-hnsw/
